cpp/dynamic_community_detection_module/algorithm/dynamic_community_detection.cpp

```diff
@@ -0,0 +1,370 @@
+#include <algorithm>
+#include <cmath>
+#include <mg_graph.hpp>
+#include <numeric>
+
+#include "dynamic_community_detection.hpp"
+
+namespace LabelRankT {
+/* #region helper_methods */
+///@brief Checks if unordered_set is subset of another unordered_set.
+///
+///@tparam T -- set element data type
+///
+///@param a -- subset candidate
+///@param b -- set against which the check is performed
+///
+///@return -- whether a is subset of b
+template <class T>
+bool IsSubset(const std::unordered_set<T>& a, const std::unordered_set<T>& b) {
+  return std::includes(a.begin(), a.end(), b.begin(), b.end());
+}
+
+std::vector<std::uint64_t> LabelRankT::NodesMemgraphIDs() const {
+  std::vector<std::uint64_t> nodes;
```

Comment:
you know how many nodes you will add so you can call [reserve](https://en.cppreference.com/w/cpp/container/vector/reserve).